source/hands_on/ICM.rst

```diff
@@ -0,0 +1,265 @@
+
+4. Reward normalization. Since the reward is unstable, it is necessary to normalize the reward to [0, 1] to make the learning more stable. Here we use the maximum and minimum normalization method.
+
+5. Feature normalization. By integrating intrinsic and extrinsic rewards, it is important to ensure that intrinsic rewards scale across different feature representations, which can be achieved through batch normalization.
+
+6. More actors (more collectors in DI-engine): Adding more parallel actors can make training more stable.
+
+Key Equations or Key Graphs
+---------------------------------------
+The overall training and calculation process of the ICM algorithm is as follows:
+
+.. image:: images/ICM_illustraion.png
+   :align: center
+   :scale: 70%
+
+1. As shown in the figure on the left, the agent samples the action a in the state :math:`s_t` through the current policy :math:`\pi` and executes it, and finally obtains the state :math:`s_{t+1}`. The total reward is the sum of two partial rewards, one part is the external reward: math:`r_t^e`, that is, the sparse reward obtained in the environment; the other part is the intrinsic reward obtained by the ICM :math:`r_t^i` (The specific calculation process is given in step 4), the final strategy needs to achieve the purpose of training by optimizing the total reward.
+The specific formula is as follows:
+
+ :math:`r_t=r_t^i + r_t^e`
+
+ :math:`{\max}_{\theta_p}\mathbb{E}_{\pi(s_t;\theta_p)}[\Sigma_t r_t]`
+
+2. In the reward module of ICM, it first extracts the eigenvectors after characterization by :math:`s_t` and :math:`s_{t+1}`,:math:`Phi(s_t; \theta_E)` and :math:`\Phi(s_{t+1}; \theta_E)` as input (reduce them to :math:`\Phi(s_t)` and :math:`\Phi(s_{t+1 })`), and output the predicted action value :math:`a_t`
+
+ :math:`\hat{a_t}=g(\Phi(s_t),\Phi(s_{t+1}) ; \theta_I)`
+
+ :math:`{\min}_{\theta_I, \theta_E} L_i(\hat{a_t},a_t)`
+
+Here :math:`\hat{a_t}` is the predicted value of :math:`a_t` and :math:`L_I` describes the difference between the two (cross entropy loss). The smaller the difference, the more accurate the estimation of the current action, and the better the representation of the environmental elements that the agent can control.
+
+
+3. The forward model of ICM will take :math:`Phi(s_t)`  and action value :math:`a_t` as input, and output the predicted value of the feature vector of the state at the next moment :math:`hat{\Phi} (s_{t+1})`

```

Comment:
3. The forward model of ICM will take :math:`\Phi(s_t)` and action value :math:`a_t` as input, and output the predicted value of the feature vector of the state at the next moment :math:`\hat{\Phi}(s_{t+1})` .